mifosng-android/src/main/java/com/mifos/mifosxdroid/online/savingaccountsummary/SavingsAccountSummaryFragment.java

```diff
@@ -456,4 +484,107 @@ public void onDestroyView() {
         void doTransaction(SavingsAccountWithAssociations savingsAccountWithAssociations, String
                 transactionType, DepositType accountType);
     }
+
+    // Takes ScreenShot of Relative layout
+    public Bitmap layoutToImage() {
+        File dir = new File(Environment.getExternalStorageDirectory().getPath(), "Mifos Docs");
+        if (!dir.isDirectory()) {
+            dir.mkdirs();
+        }
+        RelativeLayout relativeLayout = rootView.findViewById(R.id.savings_account_summary_pdf_rl);
+        relativeLayout.setDrawingCacheEnabled(true);
+        relativeLayout.setDrawingCacheBackgroundColor(getResources().getColor(R.color.white));
+        relativeLayout.buildDrawingCache();
+        Bitmap bm = relativeLayout.getDrawingCache();
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bm.compress(Bitmap.CompressFormat.JPEG, 100, bytes);
+        return bm;
+    }
+
+    //Converts image to pdf and deletes image
```

Comment:
Why this approach?

Reply:
as danish jamal said saving image, making pdf from it and deleting it was not efficient so i took this approach to reduce process.

Reply:
If this is not efficient way please guide me. Because we have to go any of this way. Making pdf directly needs tabular formatted string which will not support all screen sizes. And if other way is possible then we should go that way.

Reply:
Try using canvas

Reply:
How to formate it hardcoded spaces won't work.